cli/src/test/java/com/box/l10n/mojito/cli/command/ThirdPartySyncCommandTest.java

```diff
@@ -1,32 +1,96 @@
+import java.util.List;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
+import static com.box.l10n.mojito.rest.thirdparty.ThirdPartySyncAction.MAP_TEXTUNIT;
+import static com.box.l10n.mojito.rest.thirdparty.ThirdPartySyncAction.PUSH_SCREENSHOT;
+import static org.assertj.core.api.Assertions.assertThat;
 
 public class ThirdPartySyncCommandTest extends CLITestBase {
 
-    /**
-     * logger
-     */
-    static Logger logger = LoggerFactory.getLogger(ThirdPartySyncCommandTest.class);
+    @Autowired
+    @Qualifier("fail_on_unknown_properties_false")
+    ObjectMapper objectMapper;
+
+    @Autowired
+    Scheduler scheduler;
+
+    Matcher<JobKey> jobMatcher;
+    TestingJobListener testingJobListener;
+
+    @Before
+    public void setUp() throws SchedulerException {
+        testingJobListener = new TestingJobListener(objectMapper);
+        jobMatcher = new PollableTaskJobMatcher<>(ThirdPartySyncJob.class);
+        scheduler.getListenerManager().addJobListener(testingJobListener, jobMatcher);
+    }
+
+    @After
+    public void tearDown() throws SchedulerException {
+        scheduler.getListenerManager().removeJobListener(testingJobListener.getName());
+        scheduler.getListenerManager().removeJobListenerMatcher(testingJobListener.getName(), jobMatcher);
+    }
 
     @Test
     public void execute() throws Exception {
+
         String repoName = testIdWatcher.getEntityName("thirdpartysync_execute");
+
         Repository repository = repositoryService.createRepository(repoName, repoName + " description", null, false);
-        getL10nJCommander().run("thirdparty-sync", "-r", repository.getName(), "-p", "does-not-matter-yet", "-ps", " _");
+        String projectId = testIdWatcher.getEntityName("projectId");
+
+        // TODO: For a plural separator like " _" this test will fail. The current version we have for
+        //  JCommander trims the argument values, even when quoted.
+        // https://github.com/cbeust/jcommander/issues/417
+        // https://github.com/cbeust/jcommander/commit/4aec38b4a0ea63a8dc6f41636fa81c2ebafddc18
+        String pluralSeparator = "_";
+        String skipTextUnitPattern = "%skip_text_pattern";
+        String skipAssetPattern = "%skip_asset_pattern%";
+        List<String> options = Arrays.asList(
+                "special-option=value@of%Option",
+                "smartling-placeholder-custom=\\{\\{\\}\\}|\\{\\{?.+?\\}\\}?|\\%\\%\\(.+?\\)s|\\%\\(.+?\\)s|\\%\\(.+?\\)d|\\%\\%s|\\%s"
+        );
+
+        getL10nJCommander().run("thirdparty-sync",
+                "-r", repository.getName(),
+                "-p", projectId,
+                "-a", MAP_TEXTUNIT.name(), PUSH_SCREENSHOT.name(),
+                "-ps", pluralSeparator,
+                "-st", skipTextUnitPattern,
+                "-sa", skipAssetPattern,
+                "-o", options.get(0), options.get(1));
+
+        waitForCondition("Ensure ThirdPartySyncJob gets executed",
```

Comment:
I guess that listener is ok for now though I think changing the application context is not too great if we ever start running test in parallel, etc. 

So my preference would be avoiding the listener and just wait for the state (or different part of the state) to get to what we expect which would indicate a successful migration.

On top of my head I don't remember other test intercepting part of the request to perform input validation. In general those are more like end to end test. The parameters are not check directly but instead the results are checked inferring that the proper arguments were passed

Reply:
I've prioritized on testing that the parameters given to the CLI were passed around downstream all the way to the Quartz job for this test. I do agree that here is better to test final states or side effects, but since non of the parameters is being used in `ThirdPartyTMSInMemory` method we have nowadays and we're not triggering side effects, my preference for this PR was to have the argument passing fully tested (since that's what we're changing now).

Since we'll still implement new methods in `ThirdPartyTMS` in the upcoming PRs, I can add end to end testing through the `ThirdPartyTMSInMemory` class once we have these new method added to the interface signature as well as to the interface implementors (`ThirdPartyTMSSmartling` and `ThirdPartyTMSInMemory`)